gettsim/transfers/rente.py

```diff
@@ -274,25 +300,419 @@ def ges_rente_zugangsfaktor(
+        serves as reference for calculating gains in the Zugangsfakor in case
+        of later retirement.
+
+    Parameters
+    ----------
+    geburtsjahr
+        See basic input variable :ref:`geburtsjahr <geburtsjahr>`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+    ges_rente_vorraus_regelrente
+        See :func: ``ges_rente_vorraus_regelrente``.
+
+    Returns
+    -------
+    """
+    if ges_rente_vorraus_regelrente:
+        out = piecewise_polynomial(
+            x=geburtsjahr,
+            thresholds=ges_rente_params["regelaltersgrenze"]["thresholds"],
+            rates=ges_rente_params["regelaltersgrenze"]["rates"],
+            intercepts_at_lower_thresholds=ges_rente_params["regelaltersgrenze"][
+                "intercepts_at_lower_thresholds"
+            ],
+        )
+    else:
+        out = 9000  # arbitrary high number to represent non-feasible retirement age
+    return out
+
+
+def ges_rente_grenze_langjährig_frau(
+    geburtsjahr: int,
+    geburtsmonat: int,
+    ges_rente_params: dict,
+    ges_rente_vorraussetz_frauen: bool,
+    ges_rente_vorraussetz_langjährig: bool,
+) -> float:
+    """Calculates the age, at which a longterm insured person or women is eligible to
+        claim the full pension. Full retirement age (FRA) without deductions.
+        This age is smaller or equal to the regelaltersgrenze (FRA<=NRA). These two
+        pension schemes allow for early retirement with deductions. Hence these
+        thresholds are needed as reference for calculating the
+        zugangsfaktor.
+
+
+    Parameters
+    ----------
+    geburtsjahr
+        See basic input variable :ref:`geburtsjahr <geburtsjahr>`.
+    geburtsmonat
+        See basic input variable :ref:`geburtsmonat <geburtsmonat>`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+    ges_rente_vorraussetz_frauen
+        See :func:`ges_rente_vorraussetz_frauen`
+    ges_rente_vorraussetz_langjährig
+        See :func:``ges_rente_vorraussetz_langjährig``
+
+    Returns
+    -------
+    returns the lowest full retirement age out of the pension schemes that allow for
+    early claims with deductions.
+
+    """
+
+    if geburtsjahr < 1951:
+        x_long = geburtsjahr + (geburtsmonat - 1) / 12
+    else:
+        x_long = geburtsjahr
+
+    if ges_rente_vorraussetz_langjährig:
+        pension_for_long = piecewise_polynomial(
+            x=x_long,
+            thresholds=ges_rente_params["altersgrenze_langjährig_versicherte"][
+                "thresholds"
+            ],
+            rates=ges_rente_params["altersgrenze_langjährig_versicherte"]["rates"],
+            intercepts_at_lower_thresholds=ges_rente_params[
+                "altersgrenze_langjährig_versicherte"
+            ]["intercepts_at_lower_thresholds"],
+        )
+    else:
+        pension_for_long = (
```

Comment:
Again, I would much prefer if we got rid of the 9000 and use two variables instead:

- the altersgrenze in case `ges_rente_vorraussetz_langjährig` would be true
- eligibility `ges_rente_vorraussetz_langjährig` which we could use later to use the altersgrenze or not.